File: myLeetCode/Greedy/455. Assign Cookies.cpp
// 455. Assign Cookies
// https://leetcode.com/problems/assign-cookies/

#include <iostream>
#include <vector>
#include <algorithm>

int findContentChildren(std::vector<int>& greed, std::vector<int>& cookies) {
    std::sort(greed.begin(), greed.end());
    std::sort(cookies.begin(), cookies.end());

    int greedIndex = 0;
    int cookieIndex = 0;
    int count = 0;

    while (greedIndex < greed.size() && cookieIndex < cookies.size()) {
        if (greed[greedIndex] <= cookies[cookieIndex]) {
            count++;
            greedIndex++;
        }
        cookieIndex++;
    }

return count;
}

int main() {
    std::vector<int> greed = {1,2}; // greed factors of children
    std::vector<int> cookies = {1,2,3}; // sizes of cookies
    std::cout << findContentChildren(greed, cookies) << std::endl;

    return 0;
}

File: C++/代码随想录/Greedy/455. Assign Cookies.py
def findContentChildren(g, s):
    g.sort()
    s.sort()
    child_i = cookie_j = 0
    while child_i < len(g) and cookie_j < len(s):
        if g[child_i] <= s[cookie_j]:
            child_i += 1
        cookie_j += 1
    return child_i

# 测试用例
g = [1, 2, 3]  # 孩子的贪心指数
s = [1, 1]     # 饼干的大小
print(findContentChildren(g, s))  # 输出应该是 1
